packages/components/src/Badge/styles.tsx

```diff
@@ -3,95 +3,87 @@ import { Badge } from '@mui/material';
+  const colors = {
+    contained: {
+      grey: theme.palette.grey[800],
+      primary: theme.palette.primary[800],
+      error: theme.palette.red[800],
+      success: theme.palette.green[800],
+      warning: theme.palette.yellow[800],
+      default: theme.palette.primary.main,
+    },
+    light: {
+      grey: theme.palette.grey[300],
+      primary: theme.palette.primary[100],
+      error: theme.palette.red[100],
+      success: theme.palette.green[100],
+      warning: theme.palette.yellow[100],
+      default: theme.palette.primary.main,
+    },
+  };
 
   if (customColor === 'white') {
     return theme.palette.background.default;
   }
 
-  if (customColor === 'warning') {
-    return theme.palette.warning.main;
-  }
-
-  if (customColor === 'error') {
-    return theme.palette.error.dark;
-  }
-
-  return theme.palette.primary.main;
+  return colors[variantColor][customColor || 'default'];
 };
 
 const getTextColor = ({
   customColor,
   theme,
+  variantColor,
 }: StyledBadgeThemeProps): string => {
-  if (customColor === 'grey') {
-    return theme.palette.text.primary;
-  }
-
-  if (customColor === 'primary') {
-    return theme.palette.primary.contrastText;
-  }
+  const lightTextColor = {
+    grey: theme.palette.grey[900],
+    primary: theme.palette.primary[900],
+    error: theme.palette.red[900],
+    success: theme.palette.green[900],
+    warning: theme.palette.yellow[900],
+    default: theme.palette.primary.contrastText,
+  };
 
   if (customColor === 'white') {
     return theme.palette.grey[900];
   }
 
-  if (customColor === 'error') {
-    return theme.palette.error.contrastText;
-  }
-
-  if (customColor === 'errorLight') {
-    return theme.palette.error.dark;
-  }
-
-  if (customColor === 'success') {
-    return theme.palette.primary.contrastText;
-  }
-
-  if (customColor === 'warning') {
-    return theme.palette.error.contrastText;
+  if (variantColor === 'contained') {
+    return theme.palette.common.white;
   }
 
-  return theme.palette.primary.contrastText;
+  return lightTextColor[customColor || 'default'];
 };
 
 export const StyledBadge = styled(Badge, {
-  shouldForwardProp: (prop) => prop !== 'customColor' && prop !== 'withBorder',
+  shouldForwardProp: (prop) =>
+    prop !== 'customColor' && prop !== 'withBorder' && prop !== 'variantColor',
 })<StyledBadgeProps>`
   .MuiBadge-badge {
     height: 20px;
     padding: ${({ theme }) => theme.spacing(0, 1)};
 
     font-size: ${({ theme }) => theme.typography.small.fontSize};
     line-height: 20px;
-    color: ${({ customColor, theme }) => getTextColor({ customColor, theme })};
+    color: ${({ customColor, theme, variantColor }) =>
+      getTextColor({ customColor, theme, variantColor })};
 
-    background-color: ${({ customColor, theme }) =>
-      getBgColor({ customColor, theme })};
+    background-color: ${({ customColor, theme, variantColor }) =>
+      getBgColor({ customColor, theme, variantColor })};
```

Comment:
```suggestion
      getBgColor({ $customColor, theme, $variantColor })};
```

Пропсы, которые не должны попадать в dom, должны начинаться с $